### PR TITLE
BaseNotifier class made pydantic safe (with backward edits)

### PR DESCRIPTION
**Pull request by snwagh**
Fixes [1647](https://github.com/OpenMined/Heartbeat/issues/1647) and [1661](https://github.com/orgs/OpenMined/projects/88/views/16?pane=issue&itemId=72409982)

By making BaseNotifier and SMTPClient classes Pydantic safe

### CODE REVIEW

File: packages/syft/src/syft/service/notifier/smtp_client.py [hunk 1]
```diff
@@ -4,13 +4,14 @@
...
+from pydantic import BaseModel
 from result import Err
 from result import Ok
 from result import Result
 
 
-class SMTPClient:
-    SOCKET_TIMEOUT = 5  # seconds
+class SMTPClient(BaseModel):
+    SOCKET_TIMEOUT: float = 5  # seconds
```
Comment (tcp) [inline packages/syft/src/syft/service/notifier/smtp_client.py]: Hey @snwagh! Not `int`?

Comment (snwagh) [reply packages/syft/src/syft/service/notifier/smtp_client.py]: I found a reference that said the `timeout` parameter of smtplib was float but it seems that I was wrong. It is `int32`, will change this, thanks!

Comment (snwagh) [reply packages/syft/src/syft/service/notifier/smtp_client.py]: Resolved [4fc02d9](https://github.com/OpenMined/PySyft/pull/9107/commits/4fc02d99524abf97a58db8fa74aefcdb9cc96feb)

**Review by BrendanSchell — APPROVED** [review]
LGTM 🚀

**Pull request merged**